pkg/database/database.go

```diff
@@ -149,33 +149,39 @@ func (db *Database) OpenWithCacher(ctx context.Context, cacher cache.Cacher) err
 	// opencenusus stats.
 	var rawSQL *sql.DB
 	if err := withRetries(ctx, func(ctx context.Context) error {
-		var err error
-		rawSQL, err = sql.Open("ocsql", c.ConnectionString())
+		s, err := sql.Open("ocsql", c.ConnectionString())
 		if err != nil {
+			return fmt.Errorf("failed to open sql connection: %w", err)
```

Comment:
This is intentionally not retryable. "Open" doesn't test the connection, it just parses the string and creates a struct.